platform/core/src/services/HangingProtocolService/HangingProtocolService.js

```diff
@@ -54,6 +65,21 @@ class HangingProtocolService {
   addProtocols(protocols) {
     protocols.forEach(protocol => {
       if (this.protocols.indexOf(protocol) === -1) {
+        protocol.id = protocol.id || protocol.name;
+        protocol.name = protocol.name || protocol.id;
+        const { stages } = protocol;
+
+        stages.forEach(stage => {
```

Comment:
I know what you are doing here, but maybe put some comment of why for people reading this